common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -1554,6 +1572,10 @@ public void onFailure(String streamId, Throwable throwable) throws IOException {
         }
       }
       isWriting = false;
+      // The failed bytes (and the subsequent bytes) landed at ESS, hence we considered them
+      // also as the bytes being ignored. Including them as ignoredBlockBytes would also help
+      // measure any server side improvement.
+      updateIgnoredBlockBytes();
```

Comment:
Nit: Can we move this block to line 1559.
Also reword: `The block was received by ESS but didn't get merged so it is considered as "ignored"`.